src/components/datatablesContainer/DataTableSystems/DataTableSystems.js

```diff
@@ -678,27 +678,40 @@ export const DataTableSystems = ({
       setErrorMsgs(validationErrors);
       return false;
     }
-    try {
-      const resp = await mpApi
-        .saveSystemsComponents(
-          userInput,
-          selectedSystem.locationId,
-          selectedSystem.id,
-          selectedRangeInFirst.id
-        )
-        .catch((error) => console.log("saveSystemsComponents failed", error));
-      if (resp.status >= 200 && resp.status < 300) {
-        setupdateComponentTable(true);
-        setUpdateRelatedTables(true);
-        setErrorMsgs([]);
-        return true;
-      } else {
-        const errorResp = Array.isArray(resp) ? resp : [resp];
-        setErrorMsgs(errorResp);
-      }
-    } catch (error) {
-      setErrorMsgs([JSON.stringify(error)]);
+    
+   try {
+    let resp;
+    let response;
+
+    response = await mpApi
+    .saveComponents(
+      userInput,
+      selectedSystem.locationId,
+      selectedRangeInFirst.componentId
+    )
+    .catch((error) => console.log("saveComponents failed", error));
+
+    resp = await mpApi
+    .saveSystemsComponents(
+      userInput,
+      selectedSystem.locationId,
+      selectedSystem.id,
+      selectedRangeInFirst.id
+    )
+    .catch((error) => console.log("saveSystemsComponents failed", error));
+
+    if ((resp?.status >= 200 && resp?.status < 300) || (response?.status >= 200 && response?.status < 300)) {
+      setupdateComponentTable(true);
+      setUpdateRelatedTables(true);
+      setErrorMsgs([]);
+      return true;
+    } else {
+      const errorResp = Array.isArray(resp) ? resp : [resp];
+      setErrorMsgs(errorResp);
     }
```

Comment:
If the `saveComponents` response is non-200, it will not be printed to the user here. This should be updated to check if there are error messages on that response:
```suggestion
      const newErrorMsgs = [];
      [resp, response].forEach((res) => {
        if (res?.status < 200 || res?.status >= 300) {
          newErrorMsgs.push(res);
        }
      });
      if (newErrorMsgs.length > 0) {
        setErrorMsgs(newErrorMsgs.flat());
        return false;
      }

      setupdateComponentTable(true);
      setUpdateRelatedTables(true);
      setErrorMsgs([]);
      return true;
```

Reply:
Changes committed with last code change.